### PR TITLE
Entry type2

### PR DESCRIPTION
**Pull request by sanbrock**

### CODE REVIEW

File: src/pynxtools/nomad/parser.py [hunk 1]
```diff
@@ -491,18 +491,25 @@ def parse(
                 app_sec = getattr(self.nx_root, app)
                 try:
                     app_entry = getattr(app_sec, "ENTRY")
+                    if len(app_entry) < 1:
+                        raise AttributeError()
                 except AttributeError:
                     app_entry = getattr(app_sec, "entry")
+                    if len(app_entry) < 1:
+                        raise AttributeError()
```
Comment (RubelMozumder) [inline src/pynxtools/nomad/parser.py:499]: A quick question, app_entry is not being used anywhere. Or is it to use in the future?

Comment (sanbrock) [reply src/pynxtools/nomad/parser.py:499]: it is only to check if exists, so if no real entry is found then we do not add a corresponding AppDef to the entry_type

Comment (RubelMozumder) [reply src/pynxtools/nomad/parser.py:499]: Right! I missed line:499 Attribute error.

Comment (lukaspie) [inline src/pynxtools/nomad/parser.py:499]: Could we make this a bit cleaner? Also would be nice to have some information in the error message.
```suggestion
                for attr_name in ("ENTRY", "entry"):
                    app_entry = getattr(app_sec, attr_name, None)
                    if app_entry and len(app_entry) >= 1:
                        break
                    else:
                        raise AttributeError("No valid entry found in the NeXus file.")
```

**Review by lukaspie — CHANGES_REQUESTED** [review]
One minor comment

File: src/pynxtools/nomad/parser.py (hunk 1 above)
Comment (sanbrock) [reply src/pynxtools/nomad/parser.py:499]: The suggested code would raise exception even if ENTRY is not found, but we need to quit only if neither 'ENTRY' nor 'entry' is present and even in this case we want to catch this exception as it is not a real error at this point, but only an indication for what type of Experiment label to put into 'entry_type'.

Comment (lukaspie) [reply src/pynxtools/nomad/parser.py:499]: Ok, let's keep your version then.

**Review by lukaspie — APPROVED** [review]
lgtm

**Pull request merged**